libraries/AP_IOMCU/AP_IOMCU.cpp

```diff
@@ -371,6 +393,66 @@ void AP_IOMCU::read_rc_input()
     }
 }
 
+/*
+  read dshot erpm
+ */
+void AP_IOMCU::read_erpm()
+{
+    uint16_t *r = (uint16_t *)&dshot_erpm;
+    if (!read_registers(PAGE_RAW_DSHOT_ERPM, 0, sizeof(dshot_erpm)/2, r)) {
+        return;
+    }
+    uint8_t motor_poles = 14;
+    AP_BLHeli* blh = AP_BLHeli::get_singleton();
+    if (blh) {
+        motor_poles = blh->get_motor_poles();
+    }
+    for (uint8_t i = 0; i < IOMCU_MAX_CHANNELS/4; i++) {
+        for (uint8_t j = 0; j < 4; j++) {
+            const uint8_t esc_id = (i * 4 + j);
+            if (dshot_erpm.update_mask & 1U<<esc_id) {
```

Comment:
should update_mask be cleared at the end of this loop? It looks like it is only cleared on timeout, which means it looks like we can call update_rpm multiple times per piece of data from the IOCMU

Reply:
The first line of the function reads into dshot_erpm overwriting whatever was there before. So I don't think it needs to be called.